src/util/get-github-client-config.ts

```diff
@@ -63,13 +64,38 @@ const calculateProxyBaseUrl = async (jiraHost: string, gitHubBaseUrl: string | u
 	return envVars.PROXY;
 };
 
+const calculateApiKeyConfig = async (jiraHost: string, logger: Logger): Promise<{ apiKeyConfig: GitHubClientApiKeyConfig } | undefined> => {
+	try {
+		const maybeApiKey = await stringFlag(StringFlags.GHE_API_KEY, "", jiraHost);
+		if (maybeApiKey) {
+			logger.info("Encrypted API key found");
+			const maybeApiKeyParts = maybeApiKey.split(":");
+			const headerName = maybeApiKeyParts.shift()!;
+			const headerEncryptedValue = maybeApiKeyParts.join(":");
+			return Promise.resolve({
+				apiKeyConfig: {
+					headerName,
+					apiKeyGenerator: () =>
+						EncryptionClient.decrypt(headerEncryptedValue, {
+							jiraHost
+						})
+				}
+			});
+		}
+	} catch (err) {
+		logger.error({ err }, "Cannot calculate API key");
+	}
+	return Promise.resolve(undefined);
```

Comment:
/nit 
`return undefined` will work as well since there's an `async` in the function signature.

Reply:
Done!